python/cudf_polars/cudf_polars/dsl/expressions/base.py

```diff
@@ -249,3 +249,36 @@ def do_evaluate(
     def collect_agg(self, *, depth: int) -> AggInfo:
         """Collect information about aggregations in groupbys."""
         return AggInfo([(self, plc.aggregation.collect_list(), self)])
+
+
+class ColRef(Expr):
+    __slots__ = ("index", "table_ref")
+    _non_child = ("dtype", "index", "table_ref")
+    index: int
+    table_ref: plc.expressions.TableReference
```

Comment:
We will translate the predicate to a libcudf AST. We get the pieces referring to the left and right table separately, but want to build a single expression that represents the predicate. So we need a way of remembering which piece came from where.